### PR TITLE
feat: as group support datasource

### PR DESCRIPTION
**Pull request by deer-hang**
**What this PR does / why we need it**:
as group support datasource
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/as' TESTARGS='-run TestAccDataSourceASGroup_basic'
...
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/as -v -run TestAccDataSourceASGroup_basic -timeout 360m -parallel 4 
=== RUN   TestAccDataSourceASGroup_basic 
=== PAUSE TestAccDataSourceASGroup_basic
=== CONT  TestAccDataSourceASGroup_basic
--- PASS: TestAccDataSourceASGroup_basic (113.83s) 
PASS 
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/as        113.890s
```

### CODE REVIEW

File: docs/data-sources/as_group.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+* `instance_terminate_policy` - The instance removal policy.
+
+* `notifications` - The notification mode.
+
+* `delete_publicip` - Whether to delete the EIP bound to the ECS when deleting the ECS.
+
+* `delete_volume` - Whether to delete the data disks attached to the ECS when deleting the ECS.
+
+* `cloud_location_id` - This parameter is reserved.
```
Comment (hmbig2) [inline docs/data-sources/as_group.md]: unclear description

File: huaweicloud/services/acceptance/as/data_source_huaweicloud_as_group_test.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+		PreCheck: func() {
+			acceptance.TestAccPreCheck(t)
+		},
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceASGroup_conf(name),
+				Check: resource.ComposeTestCheckFunc(
+					dc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(dataSourceName, "scaling_group_name", name),
```
Comment (hmbig2) [inline huaweicloud/services/acceptance/as/data_source_huaweicloud_as_group_test.go]:
```suggestion
					resource.TestCheckResourceAttr(dataSourceName, "groups.0.scaling_group_name", name),
```

File: docs/data-sources/as_group.md (hunk 1 above)
Comment (deer-hang) [reply docs/data-sources/as_group.md]: this parameter is useless, i will remove it.

**Comment by deer-hang** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/as' TESTARGS='-run TestAccDataSourceASGroup_basic'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/as -v -run TestAccDataSourceASGroup_basic -timeout 360m -parallel 4 
=== RUN   TestAccDataSourceASGroup_basic 
=== PAUSE TestAccDataSourceASGroup_basic
=== CONT  TestAccDataSourceASGroup_basic
--- PASS: TestAccDataSourceASGroup_basic (114.44s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/as        114.497s
```

File: huaweicloud/services/acceptance/as/data_source_huaweicloud_as_group_test.go (hunk 2 above)
Comment (deer-hang) [reply huaweicloud/services/acceptance/as/data_source_huaweicloud_as_group_test.go]: Done

File: huaweicloud/services/as/data_source_huaweicloud_as_groups.go [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk/openstack/autoscaling/v1/groups"
+	"github.com/chnsz/golangsdk/openstack/autoscaling/v1/tags"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/helper/hashcode"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
```
Comment (hmbig2) [inline huaweicloud/services/as/data_source_huaweicloud_as_groups.go]:
```suggestion
	"fmt"
```

Comment (deer-hang) [reply huaweicloud/services/as/data_source_huaweicloud_as_groups.go]: OK

File: huaweicloud/services/as/data_source_huaweicloud_as_groups.go [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+						},
+						"health_periodic_audit_grace_period": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"instance_terminate_policy": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"notifications": {
```
Comment (ShiChangkuo) [inline huaweicloud/services/as/data_source_huaweicloud_as_groups.go]: `notifications` and `lb_listener_id` are marked as deprecated, so we can delete it directly in the data source

File: huaweicloud/services/as/data_source_huaweicloud_as_groups.go [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"id": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+								},
+							},
+						},
+						"create_time": {
```
Comment (ShiChangkuo) [inline huaweicloud/services/as/data_source_huaweicloud_as_groups.go]: please change it to `created_at`

File: huaweicloud/services/as/data_source_huaweicloud_as_groups.go [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+			log.Printf("[WARN] Error fetching tags of AS group (%s): %s", groupID, err)
+		}
+		elements = append(elements, groupMap)
+		ids = append(ids, groupID)
+	}
+
+	d.SetId(hashcode.Strings(ids))
+	mErr := multierror.Append(nil,
+		d.Set("groups", elements),
+		d.Set("region", conf.GetRegion(d)),
```
Comment (ShiChangkuo) [inline huaweicloud/services/as/data_source_huaweicloud_as_groups.go]:
```suggestion
		d.Set("region", region),
```

File: huaweicloud/services/as/data_source_huaweicloud_as_groups.go [hunk 7]
```diff
@@ -0,0 +1,321 @@
...
+	pages, err := groups.List(asClient, opts).AllPages()
+	if err != nil {
+		return diag.Errorf("error getting AS group list: %s", err)
+	}
+	groupList, err := pages.(groups.GroupPage).Extract()
+	if err != nil {
+		return diag.Errorf("error extract to AS group list: %s", err)
+	}
+
+	if len(groupList) == 0 {
```
Comment (ShiChangkuo) [inline huaweicloud/services/as/data_source_huaweicloud_as_groups.go]: we should return it with success

File: docs/data-sources/as_groups.md [hunk 8]
```diff
@@ -0,0 +1,137 @@
+---
+subcategory: "Auto Scaling"
+---
+
+# huaweicloud_as_groups
+
+Use this data source to get all AS groups.
```
Comment (ShiChangkuo) [inline docs/data-sources/as_groups.md]:
```suggestion
Use this data source to get a list of AS groups.
```

File: huaweicloud/services/as/data_source_huaweicloud_as_groups.go (hunk 4 above)
Comment (deer-hang) [reply huaweicloud/services/as/data_source_huaweicloud_as_groups.go]: done

File: huaweicloud/services/as/data_source_huaweicloud_as_groups.go (hunk 5 above)
Comment (deer-hang) [reply huaweicloud/services/as/data_source_huaweicloud_as_groups.go]: done

File: huaweicloud/services/as/data_source_huaweicloud_as_groups.go (hunk 6 above)
Comment (deer-hang) [reply huaweicloud/services/as/data_source_huaweicloud_as_groups.go]: done

File: huaweicloud/services/as/data_source_huaweicloud_as_groups.go (hunk 7 above)
Comment (deer-hang) [reply huaweicloud/services/as/data_source_huaweicloud_as_groups.go]: done

File: docs/data-sources/as_groups.md (hunk 8 above)
Comment (deer-hang) [reply docs/data-sources/as_groups.md]: done

**Comment by deer-hang** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/as' TESTARGS='-run TestAccDataSourceASGroup_basic'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/as -v -run TestAccDataSourceASGroup_basic -timeout 360m -parallel 4 
=== RUN   TestAccDataSourceASGroup_basic 
=== PAUSE TestAccDataSourceASGroup_basic
=== CONT  TestAccDataSourceASGroup_basic
--- PASS: TestAccDataSourceASGroup_basic (108.84s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/as        108.892s
```

**Comment by deer-hang** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/as' TESTARGS='-run TestAccDataSourceASGroup_basic'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/as -v -run TestAccDataSourceASGroup_basic -timeout 360m -parallel 4 
=== RUN   TestAccDataSourceASGroup_basic 
=== PAUSE TestAccDataSourceASGroup_basic 
=== CONT  TestAccDataSourceASGroup_basic
--- PASS: TestAccDataSourceASGroup_basic (94.66s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/as        94.717s
```

**Comment by deer-hang** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/as' TESTARGS='-run TestAccDataSourceASGroup_basic'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/as -v -run TestAccDataSourceASGroup_basic -timeout 360m -parallel 4 
=== RUN   TestAccDataSourceASGroup_basic 
=== PAUSE TestAccDataSourceASGroup_basic 
=== CONT  TestAccDataSourceASGroup_basic
--- PASS: TestAccDataSourceASGroup_basic (108.21s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/as        108.263s
```

**Pull request merged**